### DISCUSSION TITLE
Looking for maintainers

### DISCUSSION THREAD

**Discussion by RaddishIoW** [HaylLtd/cordova-plugin-background-geolocation#3] [General]
We are looking for people to help maintain this and the related repositories, so that the project doesn't end up being orphaned.

If you are interested in helping out in any way, from monitoring and responding to issues and PRs to working on core components, please let me know below!

**Comment by HarelM** [comment]
I'd be happy to help out maintain this fork :-)
Let me know how you would like to do this in terms of being a maintainer or sending pull requests.
Let me know what should be the package name published to npm and I might be able to test it or let our beta users test it. (I've seen issues with cordova and @company/plugin-name so I suggest to avoid it if possible).
In terms of my wish list I have a few things I wanted to do:
1. This issue: https://github.com/IsraelHikingMap/Site/issues/1427
2. Remove all the http part (if you need it then obviously this should be kept, I don't need it)
3. Support android x (I'm not sure what this means but I've seen other plugins upgrade to it...)
4. Move all code to a single repository (Currently it is split due to support for react-native I believe, which is nice but if no one will maintain the react-native wrapper then there's no real need and I think RN has its own plugin...)
5. Use github actions to automatically create and upload the plugin to npm when committing to master or when creating a tag.

In any case, I think it's a step in the right direction :-)

**Reply by RaddishIoW** [reply]
Excellent, that’s great - thank you!

Here’s how I propose we start:
I will create some “RFC” discussions to work through the most immediate concerns, including naming the package, the issues that we should prioritise, and where we should be heading a bit further into the future, and then we can make a plan and get cracking.

I’ll make you a maintainer on this repository now and set up a project board to organise things. :)

**Reply by HarelM** [reply]
Thanks for the vote of confidence :-)
I'm not sure I fully understand the RFCs you opened. I'll explain what doesn't sits right for me.
If we are planning to have a major release every now and then I would expect it to have its own RFC with the relevant version name.
I.e. if we are currently in version 1.0 I would expect to see a RFC calls Version 2.0 so that every major version would have its own discussion.
It might be the naming now since the versions are not defined. that perfectly fine.
Also I was thinking that every item I wrote above would be either its own discussion or its own issue, but I'm not sure how you would like to manage it. I mainly use issues in my repository, but I started it long before discussions were available and so I might be used to the "old" way.

I'll go with whatever works for you as long as I understand where to write what :-)

**Reply by RaddishIoW** [reply]
Yep - you were right in thinking that the RFCs were 'placeholder' really. I wanted to get a rough outline of what we needed to get in the next release, whatever that would be, and then what we wanted to see happen going forward. We could do with having a think about version numbers, as we're renaming the project/package... Maybe we should restart at 1.0.0? Or should we keep the history and make the next major version 4.0.0? I guess it doesn't make much difference actually... Any thoughts on that?

For your second point about how to deal with the issues you listed above - yep I agree that those should be "Issues" rather than "Discussions". If you wanted to go ahead and make issues for those, we can then assign them to the correct project. Sound good?

**Comment by phiasco12** [comment]
happy to help. I have limited time but I can always try to help.

**Reply by HarelM** [reply]
Great! If you have some time to look how to run the tests in CI environment it will be great as I couldn't figure it out...

**Comment by josuelmm** [comment]
Hello, I can help maintain the plugin when I can. I know very well Android, Cordova, Capacitor and Ionic.

**Reply by HarelM** [reply]
Thanks!! Any help would be appreciated. Especially in enabling the tests some how, making the CI run those tests, clean up the java tree a bit or creating an example repository.
Feel free to open issues you think are relevant or fix those that exist.

**Reply by ivinantony** [reply]
@josuelmm how to fix the permission issue in this plugin? for android 11 i have requested the app to open the setting and ass allow all the time permission but for android 8 there is no all time permission.if you have a demo code please let me know
